### PR TITLE
feat: fixes Issue #281 where StatefulSets with no VolumeClaimTemplates are being failed wrongly

### PR DESCRIPTION
**Pull request by olcuhu**
**All Submissions.**

- [X] Have you followed the guidelines in our [Contributing document](../blob/master/CONTRIBUTING.md)?
- [X] Have you checked to ensure there aren't other open [Pull Requests][pull_requests] for the same update/change?

**Code Submissions.**

- [X] Does your submission pass linting, tests, and security analysis?

**Changes to Core Features.**

- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
Scans of StatefulSets now pass if no volumeClaimTemplate is present, as per #281. This is because it is legitimate for a StatefulSet not to have a volumeclaimTemplate section. If volumeClaimTemplates do exist, they are scanned as before for the presence of storage requests & the ReadWriteOnce access mode.
- [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

**Comment by olcuhu** [comment]
Some checks may fail until PR #345 has been merged, marking as draft PR for the moment

File: test/1_cli.bats [hunk 1]
```diff
@@ -83,6 +83,12 @@ teardown() {
   assert_gt_zero_points
 }
 
+@test "passes StatefulSet with no volumeClaimTemplate" {
+  run _app "${TEST_DIR}/asset/score-1-statefulset-novolumeclaimtemplate.yml"
```
Comment (06kellyjac) [inline test/1_cli.bats:87]: if this is the correct name the yaml file needs renaming

**Comment by olcuhu** [comment]
Fixes issue #281

File: test/asset/score-1-statefulset-novolumeclaimtemplate.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      app: nginx # has to match .spec.template.metadata.labels
+  serviceName: "nginx"
+  template:
+    spec:
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
```
Comment (06kellyjac) [inline test/asset/score-1-statefulset-novolumeclaimtemplate.yml:17]:
```suggestion
          name: web

```

no newline at end of file

image

**Review by 06kellyjac — APPROVED** [review]

**Pull request merged**